transmogrifier/sources/xml/dspace_dim.py

```diff
@@ -320,6 +223,146 @@ def get_file_formats(cls, source_record: Tag) -> list[str] | None:
     def get_format(cls) -> str:
         return "electronic resource"
 
+    @classmethod
+    def get_funding_information(cls, source_record: Tag) -> list[timdex.Funder] | None:
+        return [
+            timdex.Funder(
+                funder_name=str(funding_reference.string),
+            )
+            for funding_reference in source_record.find_all(
+                "dim:field", element="description", qualifier="sponsorship", string=True
+            )
+        ] or None
+
+    @classmethod
+    def get_identifiers(cls, source_record: Tag) -> list[timdex.Identifier] | None:
+        return [
+            timdex.Identifier(
+                value=str(identifier.string),
+                kind=identifier.get("qualifier") or "Not specified",
+            )
+            for identifier in source_record.find_all(
+                "dim:field", element="identifier", string=True
+            )
+            if identifier.get("qualifier") != "citation"
+        ] or None
```

Comment:
Using this as an example, but I'm finding these easy to scan now.  I do wonder if it's because I've seen quite a few now: list comprehension + filtering clause + `or` for some values.  It's complex, but I think once you take the time to understand one, you kind of understand the rest in the transform.

When we get into orchestration, and all FMR is done, it might be worth looking at some of these and see if some helper utilities can be constructed.

Reply:
Yes! I agree that this pattern is ripe for further abstraction. Created an [issue](https://github.com/MITLibraries/transmogrifier/issues/197) for this!